src/jolt/util/feo_calib.py

```diff
@@ -0,0 +1,319 @@
+                        if offset_candidate >= FE_OFFSET_RANGE[1]:
+                            logger.warning(f"Was not able to converge within range")
+                            fe_offset = FE_OFFSET_RANGE[1]
+                            break
+                        await_set_stabilized(
+                            dev.set_frontend_offset, dev.get_frontend_offset, offset_candidate, tolerance=0.1)
+                        # Settle
+                        time.sleep(0.2)
+                        v_output_array = repeated_get(dev.get_frontend_output_voltage, repeats=5, interval=0.03)
+                        # When no significant voltage is found, assume optimal offset is reached
+                        v_output_median = statistics.median(v_output_array)
+                        if v_output_median == 0:
+                            offset_difference = offset_candidate - fe_offset
+                            fe_offset = offset_candidate
+                            break
+                        elif v_output_median < 0:
+                            logger.error(f"Negative output voltage detected. \nPlease check the hardware for defects.")
+                            raise ValueError(f"Negative output voltage detected ({v_output_median} V)")
+
+                        # The was no convergence yet, so keep going further
+
+                        # Output (V)
+                        #    |
+                        # Vm |
+                        #    |
+                        #    |⟍   | →
+                        #    |  ⟍ ↓ →
+                        #    |    ⟍
+                        #    |_____ ⟍__________
+                        #    0                  1023
+                        #    Offset
+
+                        # Since the offset difference seems somewhat exponential, and since the signal becomes more
+                        # noisy for larger operating voltages, keeping a small step size would be pointless (and slow).
+                        # Therefore make the step size dependent on the offset difference.
+                        step_size = max(1, int(offset_difference ** 0.5))
+                        offset_candidate += step_size
+                if fe_offset:
+                    logger.info(f"Converged: {fe_offset} offset")
+                    temp_measured = dev.get_cold_plate_temp()
+                    voltage_measured = dev.get_voltage()
+                    # Saving the data
+                    with open(calibration_file_tmp, "a", newline="") as file:
+                        writer = csv.writer(file, delimiter='\t')
+                        writer.writerows([[
+                            f"{target_op_temp:.2f}",
+                            f"{temp_measured:.2f}",
+                            channel.name,
+                            f"{target_op_voltage:.2f}",
+                            f"{voltage_measured:.2f}",
+                            f"{fe_offset:d}",
+                        ]])

```

Comment:
This whole block would be nicer is a separate function. 

Also, if you write only 1 row at a time, I think it'd be clearer to use `writer.writerow()`, or has this function drawbacks?